### PR TITLE
Improve type handling to work on 32-bit systems (Superseded by #20)

### LINKED ISSUE #16: Malfunctioning on 32-bit systems and inconsistent type handling
Bit sequences should be treated as `Unsigned`, but this program makes heavy use of `Int`. 
As a result, this package does not work on 32-bit systems.
Also, although it works on 64-bit systems, the type handling is inconsistent.

Is this package still maintained?　In other words, is it worth sending a PR?

### PR DESCRIPTION
**Pull request by kimikage**
Fixes #16

I do not have a clear policy on the extent to which I should make changes. At least, this is not a minimal fix.

### CODE REVIEW

File: src/Inflate.jl [hunk 1]
```diff
@@ -33,11 +33,11 @@ module Inflate
 export inflate, inflate_zlib, inflate_gzip,
        InflateStream, InflateZlibStream, InflateGzipStream
 
-# Huffman codes are internally represented by Vector{Vector{Int}},
+# Huffman codes are internally represented by Vector{Vector{UInt16}},
```
Comment (kimikage) [inline src/Inflate.jl:36]: Perhaps this should be done in a separate PR, but I think the tables of 64-bit elements are inefficient.

File: src/Inflate.jl [hunk 2]
```diff
@@ -47,31 +47,23 @@ export inflate, inflate_zlib, inflate_gzip,
...
+    [0x090:0x0ff;]] # 9-bit
+
+const fixed_distance_table = Vector{UInt16}[[], [], [], [], [0x000:0x01f;]]
 
 abstract type AbstractInflateData end
 
 mutable struct InflateData <: AbstractInflateData
     bytes::Vector{UInt8}
-    current_byte::Int
+    current_byte::UInt8
```
Comment (kimikage) [inline src/Inflate.jl:62]: I beleve that `*_byte` should be a `UInt8` type.

File: src/Inflate.jl [hunk 3]
```diff
@@ -268,67 +294,90 @@ function _inflate(data::InflateData)
...
 
+const crc_table = [make_crc_table(n) for n = 0x00:0xff]
+
+"""
+    init_crc() -> UInt32
+"""
 function init_crc()
-    if crc_table[1] == 0
-        make_crc_table()
-    end
```
Comment (kimikage) [inline src/Inflate.jl:320]: Isn't `crc_table[1]` always zero?

File: src/Inflate.jl (hunk 1 above)
Comment (GunnarFarneback) [reply src/Inflate.jl:36]: It would be useful to decouple 32-bit correctness from optimizations. The latter can be measured to verify if they are effective.

**Comment by kimikage** [comment]
See #20.
